### PR TITLE
Be more selective when adopting pods with KubernetesExecutor

### LINKED ISSUE #28071: Kubernetes logging errors - attempting to adopt taskinstance which was not specified by database
### Apache Airflow version

2.4.3

### What happened

Using following config
```
executor = CeleryKubernetesExecutor
delete_worker_pods = False
```

1. Start a few dags running in kubernetes, wait for them to complete.
2. Restart Scheduler.
3. Logs are flooded with hundreds of errors like` ERROR - attempting to adopt taskinstance which was not specified by database: TaskInstanceKey(dag_id='xxx', task_id='yyy', run_id='zzz', try_number=1, map_index=-1)`

This is problematic because:
* Our installation has thousands of dags and pods so this becomes very noisy and the adoption-process adds excessive startup-time to the scheduler, up to a minute some times.
* It's hiding actual errors with resetting orphaned tasks, something that also happens for inexplicable reasons on scheduler restart with following log: `Reset the following 6 orphaned TaskInstances`. Making such much harder to debug. The cause of them can not be easily correlated with those that were not specified by database.

The cause of these logs are the Kubernetes executor on startup loads all pods (`try_adopt_task_instances`), it then cross references them with all `RUNNING` TaskInstances loaded via `scheduler_job.adopt_or_reset_orphaned_tasks`.
For all pods where a running TI can not be found, it logs the error above - But for TIs that were already completed this is not an error, and the pods should not have to be loaded at all.

I have an idea of adding some code in the kubernetes_executor that patches in something like a `completion-acknowleged`-label whenever a pod is completed (unless `delete_worker_pods` is set). Then on startup, all pods having this label can be excluded. Is this a good idea or do you see other potential solutions? 
Another potential solution is to inside `try_adopt_task_instances` only fetch the exact pod-id specified in each task-instance, instead of listing all to later cross-reference them.

### What you think should happen instead

_No response_

### How to reproduce

_No response_

### Operating System

Ubuntu 22.04

### Versions of Apache Airflow Providers

_No response_

### Deployment

Other Docker-based deployment

### Deployment details

_No response_

### Anything else

_No response_

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/apache/airflow/blob/main/CODE_OF_CONDUCT.md)

### PR DESCRIPTION
**Pull request by jedcunningham**
When trying to adopt "resettable" TIs from SchedulerJob, we should not list out all the pods to compare against, only those that didn't succeed. This means we will get any pods that are still starting, running, or failed (meaning the TI wasn't moved to a terminal state there, and will be in out "adoptable" list).

This avoids the scenario where a dead scheduler has both a completed, successful worker, and a still running worker, causing log lines like these about the successful one:

    ERROR - attempting to adopt taskinstance which was not specified by database: TaskInstanceKey(...)

This also makes sure we only find pods with the
`kubernetes_executor=True` label for extra safety.

Closes #28071

### CODE REVIEW

File: airflow/executors/kubernetes_executor.py [hunk 1]
```diff
@@ -772,7 +773,15 @@ def try_adopt_task_instances(self, tis: Sequence[TaskInstance]) -> Sequence[Task
         kube_client: client.CoreV1Api = self.kube_client
         for scheduler_job_id in scheduler_job_ids:
             scheduler_job_id = pod_generator.make_safe_label_value(str(scheduler_job_id))
-            query_kwargs = {"label_selector": f"airflow-worker={scheduler_job_id}"}
+            # We will look for any pods owned by the no-longer-running scheduler,
```
Comment (dstandish) [inline airflow/executors/kubernetes_executor.py:783]: do we know that `scheduler_job_ids` are all not running?

File: airflow/executors/kubernetes_executor.py [hunk 2]
```diff
@@ -772,7 +773,15 @@ def try_adopt_task_instances(self, tis: Sequence[TaskInstance]) -> Sequence[Task
...
+            # but will exclude only successful pods, as those TIs will have a terminal state
+            # and not be up for adoption!
+            # Those workers that failed, however, are okay to adopt here as their TI will
+            # still be in queued.
+            query_kwargs = {
+                "field_selector": "status.phase!=Succeeded",
+                "label_selector": f"kubernetes_executor=True,airflow-worker={scheduler_job_id}",
```
Comment (dstandish) [inline airflow/executors/kubernetes_executor.py]: i'll bet you there's a selector IN but... don't care :)

File: airflow/executors/kubernetes_executor.py (hunk 1 above)
Comment (jedcunningham) [reply airflow/executors/kubernetes_executor.py:783]: Yes, if you look up a few lines, we build that from the TI's SchedulerJob asks us to try and adopt. And those are TIs are tied to [non-running SchedulerJobs](https://github.com/apache/airflow/blob/14783c63c5d36fcd70afaaebc29c2638073b7e34/airflow/jobs/scheduler_job.py#L1455).

File: airflow/executors/kubernetes_executor.py (hunk 2 above)
Comment (jedcunningham) [reply airflow/executors/kubernetes_executor.py]: I was going to say "actually, no", but turns out there is for [labels](https://kubernetes.io/docs/concepts/overview/working-with-objects/labels/#set-based-requirement) but not for [fields](https://kubernetes.io/docs/concepts/overview/working-with-objects/field-selectors/#supported-operators).

Future work!

Comment (dstandish) [reply airflow/executors/kubernetes_executor.py]: 🙃

**Review by ephraimbuddy — APPROVED** [review]

**Pull request merged**